website/project/model.py

```diff
@@ -1519,11 +1551,7 @@ def register_node(self, schema, auth, template, data):
         :param auth: All the auth information including user, API key.
         :template: Template name
         :data: Form data
-
         """
-        if not self.can_edit(auth):
```

Comment:
It'd be good to raise an error if `auth.user` does not have permissions to register the node. Something like:

```python
if not self.can_edit(auth) and not self.is_admin_parent(auth.user):
    raise PermissionsError(..)
```